### PR TITLE
[SCB-372] support user set metrics prometheus exporter address not only port

### PR DESCRIPTION
**Pull request by zhengyangyong**
Signed-off-by: zhengyangyong <yangyong.zheng@huawei.com>

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
Before:
```yaml
servicecomb:
  metrics:
    prometheus:
      port: 9696
```
Now:
```yaml
servicecomb:
  metrics:
    prometheus:
      address: localhost:9696
```

### CODE REVIEW

File: metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java [hunk 1]
```diff
@@ -37,20 +36,34 @@
...
-    int publishPort = DynamicPropertyFactory.getInstance().getIntProperty(METRICS_PROMETHEUS_PORT, 9696).get();
-    MetricsCollector metricsCollector = new MetricsCollector();
-    metricsCollector.register();
-    try {
-      this.httpServer = new HTTPServer(new InetSocketAddress(publishPort), CollectorRegistry.defaultRegistry, true);
-      LOGGER.info("Prometheus httpServer listened {}.", publishPort);
-    } catch (IOException e) {
-      throw new ServiceCombException("create http publish server failed", e);
+    this.init(
+        DynamicPropertyFactory.getInstance().getStringProperty(METRICS_PROMETHEUS_ADDRESS, "localhost:9696").get());
```
Comment (wujimin) [inline metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java]: 1.need to care for compatible?
2.is it better to be: 0.0.0.0:9696?

File: metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java [hunk 2]
```diff
@@ -37,20 +36,34 @@
...
+  }
+
+  public MetricsHttpPublisher(String address) {
+    this.init(address);
+  }
+
+  private void init(String address) {
+    String[] hostAndPort = address.split(":");
+    if (hostAndPort.length == 2) {
```
Comment (wujimin) [inline metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java]: is it better to flat the logic?

File: metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java (hunk 1 above)
Comment (WillemJiang) [reply metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java]: I think this address is for the Prometheus client to connect the server.

File: metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java [hunk 3]
```diff
@@ -37,20 +36,34 @@
 public class MetricsHttpPublisher implements ApplicationListener<ApplicationEvent> {
   private static final Logger LOGGER = LoggerFactory.getLogger(MetricsHttpPublisher.class);
 
-  private static final String METRICS_PROMETHEUS_PORT = "servicecomb.metrics.prometheus.port";
+  private static final String METRICS_PROMETHEUS_ADDRESS = "servicecomb.metrics.prometheus.address";
```
Comment (WillemJiang) [inline metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java]: Using the host and port setting could be a better way to go.

File: metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java (hunk 1 above)
Comment (zhengyangyong) [reply metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java]: Yes,Willem is right

File: metrics/metrics-integration/metrics-prometheus/src/test/java/org/apache/servicecomb/metrics/prometheus/TestMetricsHttpPublisher.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+public class TestMetricsHttpPublisher {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testBadPublishAddress() {
+    thrown.expect(ServiceCombException.class);
+    new MetricsHttpPublisher("a:b:c");
+    Assert.fail("testPublishAddress failed : a:b:c");
```
[the remark this answers is not in the document]
Comment (zhengyangyong) [reply metrics/metrics-integration/metrics-prometheus/src/test/java/org/apache/servicecomb/metrics/prometheus/TestMetricsHttpPublisher.java]: OK,thanks

File: metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java (hunk 3 above)
Comment (zhengyangyong) [reply metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java]: may better keep same as "cse.rest.adderss" ? or I will do as  your suggestion

File: metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java (hunk 2 above)
Comment (zhengyangyong) [reply metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java]: Done

File: metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java (hunk 1 above)
Comment (zhengyangyong) [reply metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java]: change to "0.0.0.0:9696" as default

**Pull request merged**